kasa/cli/discover.py

```diff
@@ -14,9 +14,17 @@
     Discover,
     UnsupportedDeviceError,
 )
-from kasa.discover import ConnectAttempt, DiscoveryResult
+from kasa.discover import (
+    NEW_DISCOVERY_REDACTORS,
+    ConnectAttempt,
+    DiscoveredRaw,
+    DiscoveryResult,
+)
 from kasa.iot.iotdevice import _extract_sys_info
+from kasa.protocols.iotprotocol import REDACTORS as IOT_REDACTORS
```

Comment:
This gives me code smell vibes, perhaps we should inverse the logic (if not in this PR, then later) by moving the redaction to be a part of the protocol classes?

Reply:
Ok will do in a subsequent PR. Are you thinking about a `classmethod` on the protocols? i.e. `IotProtocol.redact()`, `SmartProtocol.redact()`, `Discover.redact()` etc?

Reply:
I'm not sure what's the best approach, but I think the logic should be contained inside their respective classes instead of accessing their sort-of internals. But we can look into that later :+1: